serverless/src/forwarder.ts

```diff
@@ -206,9 +210,30 @@ export async function getExistingLambdaLogGroupsOnStack(cloudWatchLogs: CloudWat
   return logGroups ?? [];
 }
 
+/*
+Grab all the subscription filters from the log group
+Loop through the subscription filters, checking if any of the filters is a existing datadog-cloudformation-macro filter
+If we did not find a datadog-cloudformation-macro filter and the number of active subscription filters on the log group 
+is less than the maxAllowableLogGroupSubscriptions from AWS cloudwatch log groups then we subscribe.
+Otherwise we do not subscribe.
+*/
 export async function shouldSubscribeLogGroup(cloudWatchLogs: CloudWatchLogs, logGroupName: string) {
   const subscriptionFilters = await describeSubscriptionFilters(cloudWatchLogs, logGroupName);
-  return subscriptionFilters.length === 0;
+  let foundDatadogSubscriptionFilter = false;
```

Comment:
I think you could simplify this logic a bit:

- If there are too many filters, return false
- Iterate through the filters, if there is a Datadog one, immediately return false
- If you get this far, then you can return true

Reply:
The good news is that with your unit tests, it's easy to update the code and make sure that it still works the same way.

Reply:
A way to think about this is that you usually want to return an answer as early as possible if you can.